src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -576,7 +579,7 @@ private void OnEnvDTEProjectRemoved(EnvDTE.Project envDTEProject)
 
         private void OnEnvDTEProjectAdded(EnvDTE.Project envDTEProject)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
```

Comment:
Yes, let's try this in preview bit. still have hang risk, but might not easy to hit